src/main/java/redis/clients/jedis/DefaultJedisClientConfig.java

```diff
@@ -0,0 +1,201 @@
+package redis.clients.jedis;
+
+import javax.net.ssl.HostnameVerifier;
+import javax.net.ssl.SSLParameters;
+import javax.net.ssl.SSLSocketFactory;
+
+public final class DefaultJedisClientConfig implements JedisClientConfig {
+
+  private final int connectionTimeout;
+  private final int soTimeout;
+  private final int infiniteSoTimeout;
```

Comment:
With Java 8 (#2357) `Duration` could be used rather than `int`s for the timeouts, which would probably be a better API in the long run.

Reply:
I assume this goes for all places where `Protocol.DEFAULT_TIMEOUT` is used really, as it is now the unit for the timeouts is not obvious.

Reply:
@walles Let's merge this PR first and then we'll think about it :)